src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -322,43 +391,69 @@ private void flushBatch(Batch<T> batch) throws IOException, InterruptedException
         Map<Address, T> batchMap = batch.getMap();
 
         List<Address> successfulAddresses = Lists.newArrayListWithExpectedSize(batchMap.size());
-        Map<byte[], byte[]> successfulRows = Maps.newHashMap();
+        Map<byte[], byte[]> successfulRows = Maps.newTreeMap(UnsignedBytes.lexicographicalComparator());
 
         long nanoTimeBeforeBatch = System.nanoTime();
 
         try {
-            for (Map.Entry<Address, T> entry : batchMap.entrySet()) {
-                final Address address = entry.getKey();
-                final T op = entry.getValue();
-                final byte[] rowKey = ArrayUtils.addAll(uniqueCubeName, address.toWriteKey(idService));
-                final long nanoTimeBeforeWrite = System.nanoTime();
-
-                byte[] dbBytes;
-                switch (commitType) {
-                    case INCREMENT:
-                        long postIncr = increment(rowKey, op);
-                        dbBytes = Bytes.toBytes(postIncr);
-                        break;
-                    case READ_COMBINE_CAS:
-                        dbBytes = readCombineCas(rowKey, op);
-                        break;
-                    case OVERWRITE:
-                        overwrite(rowKey, op);
-                        dbBytes = op.serialize();
-                        break;
-                    default:
-                        throw new RuntimeException("Unsupported commit type " + commitType);
+            if (commitType.equals(CommitType.INCREMENT) && configuration.batchSize > 1) {
+                Iterable<List<Map.Entry<Address, T>>> partition = Iterables.partition(batchMap.entrySet(), configuration.batchSize);
```

Comment:
this is ultra nit, but seems like it should be `partitions`

Reply:
👍